app/controllers/profiles_controller.rb

```diff
@@ -1,9 +1,17 @@
 class ProfilesController < ApplicationController
   def show
     @groups_and_credits = SubjectGroup.find_each.map do |subject_group|
-      { subject_group:, credits: current_student.group_credits(subject_group) }
+      credits = current_student.group_credits(subject_group)
+      {
+        subject_group:,
+        credits:,
+        credits_needed_reached: credits >= subject_group.credits_needed,
+      }
     end
 
+    @groups_credits_met = @groups_and_credits.all? { |group_and_credits| group_and_credits[:credits_needed_reached] }
```

Comment:
```suggestion
    @all_group_credits_met = @groups_and_credits.all? { |group_and_credits| group_and_credits[:credits_needed_reached] }
```

Reply:
Yep I totally missed this. System tests would have catched this. Updated in 9bcf37427d0cf696c4684390573fbb2f71624f0c